src/OpenLoco/Ui/ScrollView.cpp

```diff
@@ -195,6 +195,7 @@ namespace OpenLoco::Ui::ScrollView
             if (winWidget->type == WidgetType::scrollview)
             {
                 (*scrollIndex)++;
+                break;
```

Comment:
This isn't correct. Although there are only ever 2 scrolls. In theory it should be able to handle more.

Reply:
I think it is correct, and I agree with the static analysis here, unless I'm missing something:
On line 202 the code tries to do `window->scroll_areas[*scrollIndex]`. The size of `scroll_areas` array is only 2, so the only valid index values are 0 and 1. In this method, `scrollIndex` starts at 0 on line 188. It is incremented here, meaning it'll have a value of 1. It cannot be incremented any more, otherwise it'll be outside the array bounds, so this is why the loop should terminate here.

Reply:
The intent of this bit of code is to find that index and it has been designed for more than 2 scrolls. I think just have a bounds check after this for loop fixing it to 0 or 1. Then at least it's clear that the issue is we can't handle more than 2 scrolls.

Reply:
Fair enough, that works for me

Reply:
fixed